enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,964 @@
+
+![Alt text](assets/new-community-operators.png?raw=true "community-operators")
+
+Since each bundle in the index belongs to a channel in a package, the index itself can be represented as a collection of packages. Each package is then represented as a json/yaml, a config file that will live inside the index container image alongside the individual bundle metadata. 
+
+#### Contents of the config file
+
+The config file for each package will have a stream of json objects, representing the package and bundle information for that package. Currently, this information is stored in a sql database inside the index. To capture all the information previously stored in this database but in a normalized way, the config file will have two kinds of json blob: 
+
+1. The json blob capturing package information. 
+```json
+{
+    "schema": "package.v1",
+    "name": "<package-name>",
+    "defaultChannel": "<channel-name>",
+    "icon": "embedded or remote",
+    "channels": ["<list-of-channels-in-package>"],
+    "description": "A message about the operator, a description of channels..."
+}
+```
+This information is currently captured in the `package` table.
+
+2. The json blob capturing the bundle information.
+```json
+{
+    "schema": "bundle.v1",
+    "name": "<operatorbundle_name>",
+    "package": "<package_name>",
+    "image": "<operatorbundle_path>",
+    "version": "<version>",
+    "properties":["<list of properties of bundle that encode bundle dependencies(provided and required apis)>"],
+    "channels": ["list of channels the bundle belongs to, which each channel represented as json blob that encodes channel name and upgrade graph information
+    
+            {
+            name: channel-name,
+            upgradeInfo: {
+                replaces: operatorbundle_name
+                skiprange: range
+                skips : list of versions  
+            }
+        }
+    
+    "],
+    "relatedImages" : ["<list-of-related-images>"]
+}
+
+```
+
+This information is currently captured in the `operatorbundle`, `properties`, `channel`, `channel_entry`, `api`, `api_provider`, `api_requirer` and `related_image` tables in the sql database.  
+
+#### Representing the upgrade graph in the channel json blob
+
+Currently, a bundle can be added into the index using `opm index add --bundles <list-of-bundle-paths> --mode replaces|semver|semver-skippatch --tag=<index-image-tag>` where the bundle images (like `quay.io/operatorhubio/etcd:v0.9.0`) are included in `<list-of-bundle-path>`. The bundle can also mention bundles it can be upgrade from using the `skips` or `skipsRange` fields in the bundle ClusterServiceVersion. With all of these information provided, the upgrade graph of the bundles in the package is calculated and stored in the `channel_entry` table of the sql database that is built inside the index, while the `skips`/`skiprange` information is persisted in the `operatorbundle` table.  
```

Comment:
Just wanted to note that channel_entry is always authoritative here. The operatorbundle table persists the values of skip/skiprange/replaces when the bundle was first unpacked/added to the index, and other index operations could have changed the real graph in channel_entry.

Reply:
Added that note